### PR TITLE
added module for reading admins, streets, pois from osm

### PR DESCRIPTION
**Pull request by datanel**
- added module for reading admins, streets, pois from osm and keeping osm2mimir command as small as possible
- applied cargo fmt 

```
src
├── admin_geofinder.rs
├── bin
│   ├── bano2mimir.rs
│   ├── bragi.rs
│   └── osm2mimir.rs (functions are now in osm_reader modules)
├── boundaries.rs
├── lib.rs
└── osm_reader (NEW)
    ├── utils.rs
    ├── mod.rs
    ├── osm_admin_reader.rs
    ├── osm_poi_reader.rs
    └── osm_street_reader.rs
```

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -39,3 +39,7 @@ extern crate geo;
 
 pub mod boundaries;
 pub mod admin_geofinder;
+pub mod osm_reader;
+
+#[macro_use]
+extern crate mdo;
```
Comment (TeXitoi) [inline src/lib.rs]: please group extern crate together.

File: src/osm_reader/mod.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+use std::fs::File;
+use std::rc::Rc;
+use std::path::Path;
+
+pub mod utils;
+pub mod osm_admin_reader;
+pub mod osm_poi_reader;
+pub mod osm_street_reader;
+
+pub type AdminsVec = Vec<Rc<mimir::Admin>>;
```
Comment (TeXitoi) [inline src/osm_reader/mod.rs]: I'd put that in osm_admin_reader.rs

File: src/osm_reader/mod.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+extern crate mimir;
+
+use std::fs::File;
+use std::rc::Rc;
+use std::path::Path;
+
+pub mod utils;
+pub mod osm_admin_reader;
+pub mod osm_poi_reader;
+pub mod osm_street_reader;
```
Comment (TeXitoi) [inline src/osm_reader/mod.rs]: We are in the osm_reader module. We can remove osm and reader from the name of these modules.

File: src/osm_reader/utils.rs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+use super::AdminsVec;
+
+pub fn make_admin_geofinder(admins: &AdminsVec) -> AdminGeoFinder {
+    let mut geofinder = AdminGeoFinder::new();
+
+    for a in admins {
+        geofinder.insert(a.clone());
+    }
+    geofinder
+}
```
Comment (TeXitoi) [inline src/osm_reader/utils.rs]: Better to do implement FromIterator for AdminGeoFinder.

File: src/osm_reader/mod.rs (hunk 2 above)
Comment (datanel) [inline src/osm_reader/mod.rs]: yes but osm_{poi, street}_reader.rs need it too

Comment (datanel) [inline src/osm_reader/mod.rs]: and utils.rs

File: src/osm_reader/utils.rs (hunk 4 above)
Comment (datanel) [inline src/osm_reader/utils.rs]: https://github.com/CanalTP/mimirsbrunn/issues/96

**Pull request merged**